c2corg_ui/static/js/searchfilters.js

```diff
@@ -0,0 +1,144 @@
+goog.provide('app.searchFiltersDirective');
+goog.provide('app.SearchFiltersController');
+
+goog.require('app');
+
+
+/**
+ * Provides the 'searchFiltersDirective' directive, which is used to enrich
+ * filtering components in the documents listing pages.
+ * 
+ * @return {angular.Directive} The directive specs.
+ * @ngInject
+ */
+app.searchFiltersDirective = function() {
+  return {
+    restrict: 'A',
+    controller: 'appSearchFiltersController',
+    bindToController: true,
+    scope: true,
+    controllerAs: 'searchFiltersCtrl',
+    link: function(scope, element, attrs, ctrl) {
+      // Init sliders
+      $('.range-between').slider({min: 0, max: 9999, value: [0, 9999]});
+      
+      var elevationFilter = $('#elevation-filter');
+      var heightFilter = $('#height-filter');
+
+      // Update elevation slider values
+      elevationFilter.on('slide', function(slideEvt) {
+        ctrl.elevation.min = slideEvt.value[0];
+        ctrl.elevation.max = slideEvt.value[1];
+        scope.$apply();
+      });
+
+      // Update height diff slider values
+      heightFilter.on('slide', function(slideEvt) {
+        ctrl.height_diff_up.min = slideEvt.value[0];
+        ctrl.height_diff_up.max = slideEvt.value[1];
+        scope.$apply();
+      });
+    }
+  }
+};
+
+
+app.module.directive('appSearchFilters', app.searchFiltersDirective);
+
+
+/**
+ * @param {angular.Scope} $scope Scope.
+ * @param {angular.JQLite} $element Element.
+ * @param {angular.Attributes} $attrs Attributes.
+ * @constructor
+ * @ngInject
+ * @export
+ */
+app.SearchFiltersController = function($scope, $element, $attrs) {
+
+  /**
+   * @type {angular.Scope}
+   * @private
+   */
+  this.scope_ = $scope;
+
+  // FIXME: not generic to all document types
+  // Init filters object
+  angular.extend(this.scope_,  {
```

Comment:
There are several (good) reasons to access the value through the controller.
In short it is clearer and avoids some Javascript pitfals. See the Angular code style for details:
https://google.github.io/styleguide/angularjs-google-style.html#controllers

> Why? Putting methods and properties directly onto the controller, instead of building up a scope object, fits better with the Google Closure class style. Additionally, using 'controller as' makes it obvious which controller you are accessing when multiple controllers apply to an element. Since there is always a '.' in the bindings, you don't have to worry about prototypal inheritance masking primitives.